watsong/jukebox.py

```diff
@@ -27,15 +35,18 @@ def jukebox() -> Any:
 
         if query:
             album_descs, err = watson.get_albums(query)
+
             if err is not None:
                 flash(str(err))
                 return render_template("jukebox.html", songs=songs, dials=DIALS)
-            spotify.cache(album_descs)
 
-            songs = spotify.get_songs(album_descs)
+            if not current_app.testing:
+                spotify.cache(album_descs, current_app.spotify)
+
+            songs = spotify.get_songs(album_descs, current_app.spotify)
 
             random.shuffle(songs)
-            songs = spotify.add_audio_features(songs)
+            songs = spotify.add_audio_features(songs, current_app.spotify)
```

Comment:
current_app.spotify is a mock when current_app.testing is true. 

I don't know if it's best to pass this to each function, or if spotify should just import current_app from flask. I chose this so that you can pass a different spotify client when running unit tests.